doc/book/cookbook/using-middleware-in-dispatch-event.md

```diff
@@ -0,0 +1,41 @@
+# Using Middleware in "dispatch" Event
+
+During Mvc Workflow, you can use the Middleware in 'dispatch' event by provide `Psr7Bridge`. For example, you have `AuthorizationMiddleware`:
+
+```php
+namespace Application\Middleware;
+
+class AuthorizationMiddleware
+{
+    public function __invoke($request, $response, $next = null)
+    {
+        // handle authorization here...
+    }
+}
+```
+
+As the request and response in 'dispatch' event is a `Zend\Http` Request and Response object, we need the bridge to convert into PSR-7 Request and Response. To do that, you can do the following:
+
+```php
+namespace Application;
+
+use Application\Middleware\AuthorizationMiddleware;
+use Zend\Psr7Bridge\Psr7ServerRequest;
+use Zend\Psr7Bridge\Psr7Response;
+
+class Module
+{
+    public function onBootstrap($e)
+    {
+        $app          = $e->getApplication();
+        $eventManager = $app->getEventManager();
+        $services     = $app->getServiceManager();
+
+        $eventManager->attach('dispatch', function ($e) use ($services) {
+            $request  = Psr7ServerRequest::fromZend($e->getRequest());
+            $response = Psr7Response::fromZend($e->getResponse());
+            $result   = ($services->get(AuthorizationMiddleware::class))($request, $response);
```

Comment:
I think that you could pass an empty closure as third parameter to the middleware, so that if the middleware `return $next($req, $res)`, then $result will be `null`, and if the middleware `return $response`, then `$result` will be a response. In case it returns a response, you can convert it back to Zend\Http and return it (short-circuiting).